presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -368,66 +369,83 @@ protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
             TableHandle tableHandle = metadata.getTableHandle(session, table)
                     .orElseThrow(() -> new SemanticException(MISSING_TABLE, showPartitions, "Table '%s' does not exist", table));
 
-            List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.empty());
-            if (layouts.size() != 1) {
-                throw new SemanticException(NOT_SUPPORTED, showPartitions, "Table does not have exactly one layout: %s", table);
+            Optional<TableLayout> layout = getLayout(showPartitions, showPartitions.getTable(), showPartitions.getWhere());
+            if (!layout.isPresent()) {
+                return simpleQuery(
```

Comment:
Theoretically this should be possible (easy to repro case), but somehow I could not get it. Is it dead code? Am I doing something wrong?

Reply:
I think we should leave the original check. To trigger it, you'll need a connector which returns multiple layouts, which I don't think any of the open source connectors do (we have an internal connector that does).

Reply:
It's really just a sanity check. `SHOW PARTITIONS` is really only useful for the Hive connector, or connectors that do partitioning exactly like Hive, and never should have existed in the first place.